internal/mesh/internal/controllers/sidecarproxy/builder/local_app.go

```diff
@@ -35,6 +40,172 @@ func (b *Builder) BuildLocalApp(workload *pbcatalog.Workload) *Builder {
 	return b
 }
 
+func buildTrafficPermissions(trustDomain string, workload *pbcatalog.Workload, computed *pbauth.ComputedTrafficPermissions) map[string]*pbproxystate.TrafficPermissions {
+	portsWithProtocol := workload.GetPortsByProtocol()
+
+	out := make(map[string]*pbproxystate.TrafficPermissions)
+	portToProtocol := make(map[string]pbcatalog.Protocol)
+	var allPorts []string
+	for protocol, ports := range portsWithProtocol {
+		for _, p := range ports {
+			allPorts = append(allPorts, p)
```

Comment:
wouldn't this have duplicates?

Reply:
When would that happen? Unless I'm missing something, each port should be included once in the values of `portsWithProtocol`.

Reply:
Oh yeah you're right! This should not happen.